### PR TITLE
Update selectorAllParse to return an Array instead of a NodeList

### PR DESCRIPTION
**Pull request by darkwing**
**Description:**
Fixes https://github.com/aframevr/aframe/issues/1129

I suspect this one will require much more work.  I've run tests and all passes but things to consider:

*  Is this a breaking change moving forward where API docs must be updated?  Is it work the breaking change?
*  What other components may need updating to accommodate an array instead of a NodeList?

### CODE REVIEW

File: src/core/propertyTypes.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ function selectorParse (value) {
 function selectorAllParse (value) {
   if (!value) { return null; }
   if (typeof value !== 'string') { return value; }
```
Comment (darkwing) [inline src/core/propertyTypes.js]: Should `value` also be wrapped in `Array.from()`?  What is the use case?  In case a `NodeList` instance is passed in?

Comment (dmarcos) [inline src/core/propertyTypes.js]: We might want to return `[]` in the cases above?

Comment (ngokevin) [inline src/core/propertyTypes.js]: I think the first case may be for `{default: ''}` so we don't need to run that.

The second case is if an array or NodeList is passed in yeah.

**Comment by ngokevin** [comment]
looks good to me

**Comment by ngokevin** [comment]
@darkwing thanks!

**Pull request merged**